src/io/tradetrust/certificate-of-origin/1.0/schema.test.ts

```diff
@@ -1,172 +1,76 @@
-        },
-      ]
-    `);
-  });
+    const mergedDocument = {
+      ...CertificateOfOriginSchema,
+      ...modifiedData
+    } as JsonLdDocument;
 
-  it("should fail when document does not have exporter's iD, name, postalAddress", () => {
-    const document = {
-      ...sampleDocument,
-      supplyChainConsignment: {
-        ...sampleDocument.supplyChainConsignment,
-        exporter: {}
-      }
-    };
-    expect(validator(document)).toBe(false);
-    expect(validator.errors).toMatchInlineSnapshot(`
-      Array [
-        Object {
-          "dataPath": ".supplyChainConsignment.exporter",
-          "keyword": "required",
-          "message": "should have required property 'iD'",
-          "params": Object {
-            "missingProperty": "iD",
-          },
-          "schemaPath": "#/allOf/1/required",
-        },
-        Object {
-          "dataPath": ".supplyChainConsignment.exporter",
-          "keyword": "required",
-          "message": "should have required property 'name'",
-          "params": Object {
-            "missingProperty": "name",
-          },
-          "schemaPath": "#/allOf/1/required",
-        },
-        Object {
-          "dataPath": ".supplyChainConsignment.exporter",
-          "keyword": "required",
-          "message": "should have required property 'postalAddress'",
-          "params": Object {
-            "missingProperty": "postalAddress",
-          },
-          "schemaPath": "#/allOf/1/required",
-        },
-      ]
-    `);
+    return expandDocument(mergedDocument).catch(e => {
+      expect(e.message).toMatchInlineSnapshot(
+        `"\\"The property invalidCOOProperties in the input was not defined in the context\\""`
+      );
+    });
   });
+});
 
-  it("should fail when document does not have importer", () => {
-    const document = omit(cloneDeep(sampleDocument), "supplyChainConsignment.importer");
-
-    expect(validator(document)).toBe(false);
-    expect(validator.errors).toMatchInlineSnapshot(`
-      Array [
-        Object {
-          "dataPath": ".supplyChainConsignment",
-          "keyword": "required",
-          "message": "should have required property 'importer'",
-          "params": Object {
-            "missingProperty": "importer",
-          },
-          "schemaPath": "#/properties/supplyChainConsignment/required",
-        },
-      ]
-    `);
+describe("invoice schema", () => {
+  it("should work with valid invoice", async () => {
+    const mergedDocument = {
+      ...InvoiceSchema,
+      ...InvoiceSampleData
+    } as JsonLdDocument;
+    expect(await expandDocument(mergedDocument)).toBeTruthy();
   });
 
-  it("should fail when document does not have importer's iD, name, postalAddress", () => {
-    const document = {
-      ...sampleDocument,
-      supplyChainConsignment: {
-        ...sampleDocument.supplyChainConsignment,
-        importer: {}
-      }
+  it("should throw error when unknown context or properties found", async () => {
```

Comment:
what are we testing exactly? shouldn't be X or Y

Reply:
done.